sdk/core/core-client-rest/src/helpers/isBinaryBody.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { isReadableStream } from "./isReadableStream.js";
+
+export function isBinaryBody(
+  body: unknown,
+): body is
+  | Uint8Array
+  | NodeJS.ReadableStream
+  | ReadableStream<Uint8Array>
+  | (() => NodeJS.ReadableStream)
```

Comment:
just curious when the body would be `(() => NodeJS.ReadableStream)`?

Reply:
This allows passing a resettable stream, which is a valid body type for core-rest-pipeline. I guess we don't do this in codegen at the moment, but I would like to in future.